diff --git a/include/rtc/description.hpp b/include/rtc/description.hpp
index 0d0c58b5..e7d6a4c9 100644
--- a/include/rtc/description.hpp
+++ b/include/rtc/description.hpp
@@ -62,6 +62,7 @@ public:
 	std::vector<string> iceOptions() const;
 	optional<string> iceUfrag() const;
 	optional<string> icePwd() const;
+	optional<string> icePacing() const;
 	optional<CertificateFingerprint> fingerprint() const;
 	bool ended() const;
 
@@ -301,7 +302,7 @@ private:
 	string mUsername;
 	string mSessionId;
 	std::vector<string> mIceOptions;
-	optional<string> mIceUfrag, mIcePwd;
+	optional<string> mIceUfrag, mIcePwd, mIcePacing;
 	optional<CertificateFingerprint> mFingerprint;
 	std::vector<string> mAttributes; // other attributes
 
diff --git a/src/description.cpp b/src/description.cpp
index b2165a89..12eac883 100644
--- a/src/description.cpp
+++ b/src/description.cpp
@@ -153,6 +153,12 @@ Description::Description(const string &sdp, Type type, Role role)
 				// takes precedence.
 				if (!mIcePwd || index == 0) // media-level for first media overrides session-level
 					mIcePwd = value;
+			} else if (key == "ice-pacing") {
+				// RFC: 8839: The "ice-pacing" is a session-level attribute that indicates the desired 
+				// connectivity-check pacing (Ta interval), in milliseconds, that the sender wishes to use.
+				// If absent in an offer or answer, the default value of the attribute is 50 ms, which 
+				// is the recommended value specified in [RFC8445].
+				mIcePacing = value;
 			} else if (key == "ice-options") {
 				// RFC 8839: The "ice-options" attribute is a session-level and media-level
 				// attribute.
@@ -207,6 +213,8 @@ std::vector<string> Description::iceOptions() const { return mIceOptions; }
 
 optional<string> Description::icePwd() const { return mIcePwd; }
 
+optional<string> Description::icePacing() const { return mIcePacing; }
+
 optional<CertificateFingerprint> Description::fingerprint() const { return mFingerprint; }
 
 bool Description::ended() const { return mEnded; }
@@ -314,6 +322,8 @@ string Description::generateSdp(string_view eol) const {
 		sdp << "a=ice-ufrag:" << *mIceUfrag << eol;
 	if (mIcePwd)
 		sdp << "a=ice-pwd:" << *mIcePwd << eol;
+	if (mIcePacing)
+		sdp << "a=ice-pacing:" << *mIcePacing << eol;
 	if (!mIceOptions.empty())
 		sdp << "a=ice-options:" << utils::implode(mIceOptions, ',') << eol;
 	if (mFingerprint)
@@ -379,6 +389,8 @@ string Description::generateApplicationSdp(string_view eol) const {
 		sdp << "a=ice-ufrag:" << *mIceUfrag << eol;
 	if (mIcePwd)
 		sdp << "a=ice-pwd:" << *mIcePwd << eol;
+	if (mIcePacing)
+		sdp << "a=ice-pacing:" << *mIcePwd << eol;
 	if (!mIceOptions.empty())
 		sdp << "a=ice-options:" << utils::implode(mIceOptions, ',') << eol;
 	if (mFingerprint)
